### DISCUSSION TITLE
Is it possible to use environment variable for deployment_name field in flow.dag.yaml?

### DISCUSSION THREAD

**Discussion by niveditau** [microsoft/promptflow#2262] [General]
Is it possible to use environment variable for deployment_name field in flow.dag.yaml? If possible from .env or environment.yaml file.

**Comment by D-W-** [comment]
Hi @niveditau , currently it's supported to write environment variables in flow.dag.yaml like this: https://microsoft.github.io/promptflow/how-to-guides/faq.html#how-to-configure-environment-variables
But .env or environment.yaml file are not supported.

**Reply by niveditau** [reply]
Thanks!

**Reply by niveditau** [reply]
@D-W- I tried following:

1. Added this to my flow.dag.yaml
`environment_variables:  
      DEPLOYMENT_NAME: my_deployment`
2. Then this is how I am referencing environment variable in my flow.dag.yaml
`inputs:
     deployment_name: DEPLOYMENT_NAME`
3. Then I use following command to pass the environment variable
`--environment-variable DEPLOYMENT_NAME="my_new_deployment"`

And I am getting "DeploymentNotFound" error. Can you point out what mistake I am making.

**Reply by niveditau** [reply]
If you could share the flow.dag.yaml showing how we reference the environment variable (step number 2 in the above message) that would be great. Thanks!

**Reply by niveditau** [reply]
For more context I am passing environment variable with `pf flow serve` command.

**Reply by D-W-** [reply]
Hi @niveditau , referencing environment variable by name in flow/node inputs is not supported (#2 above). You can reference the set environment variable with code like this:

```python
import os

deployment_name = os.environ.get('DEPLOYMENT_NAME') 
```
